### PR TITLE
Address CWA 2022 002

### PR DESCRIPTION
**Pull request by the-frey**
Bump wasmvm to patch vulnerability

### CODE REVIEW

**Review by faddat — APPROVED** [review]
:).

This looks great and I've test-compiled on my machine as well

**Comment by webmaster128** [comment]
See
- https://github.com/CosmWasm/wasmd/blob/master/Dockerfile
- https://github.com/CosmWasm/wasmd/pull/790
- https://github.com/CosmWasm/wasmd/pull/800

for the changes needed to work with the latest static library. Also there is https://github.com/CosmWasm/wasmvm/wiki/Linking-wasmvm_muslc to explain the advanced linking required from beta9 on.

**Comment by the-frey** [comment]
Yep, just spotted the dockerfile issue. Also noticed you've coined a new wasmd, so will just bump 0.24 to that version. Thanks @webmaster128

**Comment by the-frey** [comment]
Aha, of course I need to update the CI steps as well. Probably the docker packages too. Right

File: Dockerfile [hunk 1]
```diff
@@ -15,11 +15,18 @@ WORKDIR /code
...
 # See https://github.com/CosmWasm/wasmvm/releases
-ADD https://github.com/CosmWasm/wasmvm/releases/download/v1.0.0-beta7/libwasmvm_muslc.a /lib/libwasmvm_muslc.a
-RUN sha256sum /lib/libwasmvm_muslc.a | grep d0152067a5609bfdfb3f0d5d6c0f2760f79d5f2cd7fd8513cafa9932d22eb350
+ADD https://github.com/CosmWasm/wasmvm/releases/download/v1.0.0-beta10/libwasmvm_muslc.aarch64.a /lib/libwasmvm_muslc.aarch64.a
+ADD https://github.com/CosmWasm/wasmvm/releases/download/v1.0.0-beta10/libwasmvm_muslc.x86_64.a /lib/libwasmvm_muslc.x86_64.a
+RUN sha256sum /lib/libwasmvm_muslc.aarch64.a | grep 5b7abfdd307568f5339e2bea1523a6aa767cf57d6a8c72bc813476d790918e44
+RUN sha256sum /lib/libwasmvm_muslc.x86_64.a | grep 2f44efa9c6c1cda138bd1f46d8d53c5ebfe1f4a53cf3457b01db86472c4917ac
+
+# Copy the library you want to the final location that will be found by the linker flag `-lwasmvm_muslc`
+RUN cp /lib/libwasmvm_muslc.${arch}.a /lib/libwasmvm_muslc.a
```
Comment (webmaster128) [inline Dockerfile:24]: This `${arch}` variable needs to be declared in the top of the file (see wasmd)

**Comment by webmaster128** [comment]
Also check https://github.com/CosmWasm/wasmd/wiki/Maintainer-notes for how to do the different docker builds in the multi-architecture world.

**Review by JakeHartnell — APPROVED** [review]

**Comment by giansalex** [comment]
when deleting the dockerfile there will also be an error in [release ci](https://github.com/CosmosContracts/juno/blob/main/.github/workflows/release.yml)

**Comment by the-frey** [comment]
Reinstating Dockerfile in #179

**Review by faddat — COMMENTED** [review]
so we regressed after this?

**Comment by webmaster128** [comment]
> so we regressed after this?

This PR installs beta7, not beta10 because beta7 > beta10. See `go list -m github.com/CosmWasm/wasmvm` and 

```
$ go list -m -versions github.com/CosmWasm/wasmvm
github.com/CosmWasm/wasmvm v0.1.0 v0.2.0 v0.3.0 v0.3.1 v0.3.2 v0.3.3 v0.4.0 v0.4.1 v0.6.0 v0.6.1 v0.6.2 v0.6.3 v0.7.0 v0.7.1 v0.7.2 v0.8.0-alpha2 v0.8.0-alpha3 v0.8.0 v0.8.1 v0.9.0-alpha2 v0.9.0-alpha3 v0.9.0-beta v0.9.0 v0.9.1 v0.9.4 v0.10.0-alpha2 v0.10.0-alpha3 v0.10.0-alpha4 v0.10.0 v0.11.0-alpha1 v0.11.0-alpha2 v0.11.0-rc v0.11.0 v0.12.0-alpha1 v0.12.0-alpha2 v0.12.0-alpha3 v0.12.0 v0.13.0 v0.13.1 v0.14.0-alpha1 v0.14.0-alpha2 v0.14.0-beta1 v0.14.0-beta2 v0.14.0-beta3 v0.14.0-beta4 v0.14.0-beta5 v0.14.0-rc1 v0.14.0 v0.15.0-rc1 v0.15.0 v0.15.1 v0.16.0-rc1 v0.16.0-rc2 v0.16.0-rc3 v0.16.0 v0.16.1 v0.16.2 v0.16.3 v0.16.5 v0.16.6 v0.16.7 v1.0.0-beta v1.0.0-beta10 v1.0.0-beta2 v1.0.0-beta3 v1.0.0-beta4 v1.0.0-beta5 v1.0.0-beta6 v1.0.0-beta7 v1.0.0-beta8 v1.0.0-beta9 v1.0.0-soon v1.0.0-soon2
```

**Pull request merged**